### PR TITLE
fixed admin archive channel

### PR DESCRIPTION
**Pull request by mosheBelostotsky**

### CODE REVIEW

File: lib/advanced_adapter/adapter-slack.coffee [hunk 1]
```diff
@@ -139,8 +139,8 @@ class Adapter
       opts.channel = r
       return SlackApi.channels.rename(opts)
     .then (r) ->
-      ret.push(new Channel(channel.id, channel.name, channel.name,
-        channel.created, ''))
+      return new Channel(r.id, r.name, r.name,
```
Comment (eitanshp) [inline lib/advanced_adapter/adapter-slack.coffee]: Please consolidate into a single line

File: src/admin.coffee [hunk 2]
```diff
@@ -31,16 +31,14 @@ module.exports = (robot) ->
...
-      msg.reply 'channel name should start with #'
-      return
-    else if (!(channel = /(#[^\s]+)/i.exec(msg.envelope.message.text)))
-      msg.reply 'could not find channel '+msg.match[1]
-      return
-    channel = channel[1]
-    _robot.logger.debug 'archiving channel: '+channel
+    else if msg.match[1]=='this'
+      channel = msg.envelope.message.room
+      if channel[0] == 'D'
```
Comment (eitanshp) [inline src/admin.coffee]: Please remove this check and pass it back to the collab. tool API. Slack shouldn't be different than any other collab. tool and if they decide to change to 'P' tomorrow (for private) we don't want to chase them with code changes.

File: lib/advanced_adapter/adapter-slack.coffee [hunk 3]
```diff
@@ -139,8 +139,8 @@ class Adapter
       opts.channel = r
```
Comment (eitanshp) [inline lib/advanced_adapter/adapter-slack.coffee]: Please give a more meaningful name than 'r'... (throughout the code...)

File: lib/advanced_adapter/adapter-slack.coffee (hunk 1 above)
Comment (eitanshp) [inline lib/advanced_adapter/adapter-slack.coffee]: Please give a more meaningful name than 'r'... (throughout the code...)

**Comment by gmatoshp** [comment]
looks good.

**Pull request merged**